apps/meteor/app/livechat/server/api/lib/users.ts

```diff
@@ -10,18 +11,39 @@ import { escapeRegExp } from '@rocket.chat/string-helpers';
 async function findUsers({
 	role,
 	text,
+	onlyAvailable = false,
+	excludeId,
+	showIdleAgents = true,
 	pagination: { offset, count, sort },
 }: {
 	role: IRole['_id'];
 	text?: string;
+	onlyAvailable?: boolean;
+	excludeId?: string;
+	showIdleAgents?: boolean;
 	pagination: { offset: number; count: number; sort: any };
 }): Promise<{ users: ILivechatAgent[]; count: number; offset: number; total: number }> {
-	const query = {};
+	const query: FilterOperators<ILivechatAgent> = {};
+	const orConditions: FilterOperators<ILivechatAgent>['$or'] = [];
 	if (text) {
 		const filterReg = new RegExp(escapeRegExp(text), 'i');
-		Object.assign(query, {
-			$or: [{ username: filterReg }, { name: filterReg }, { 'emails.address': filterReg }],
-		});
+		orConditions.push({ $or: [{ username: filterReg }, { name: filterReg }, { 'emails.address': filterReg }] });
+	}
+
+	if (onlyAvailable) {
+		query.statusLivechat = 'available';
+	}
+
+	if (excludeId) {
+		query._id = { $ne: excludeId };
```

Comment:
```suggestion
		query._id = { $ne: String(excludeId) };
```

The type of excludeId is not being validated anywhere.

Reply:
Type's validated by `ajv` at endpoint level.
```
		excludeId: {
			type: 'string',
			nullable: true,
		},
```

Reply:
Ah, sorry, I searched on the wrong branch :facepalm: